content/posts/batching-requests.md

```diff
@@ -0,0 +1,186 @@
+---
+title: "Batching Requests in APISIX Ingress"
+date: 2023-12-01T10:13:43+05:30
+draft: true
+ShowToc: true
+TocOpen: true
+ShowRelatedContent: false
+summary: "A tutorial on using APISIX's request batching capability in Kubernetes Ingress."
+tags: ["ingress", "kubernetes", "apache apisix", "cloud-native"]
+categories: ["API Gateway"]
+series: ["Hands-On With Apache APISIX Ingress"]
+cover:
+  image: "/images/batching-requests/candy-banner.jpg"
+  alt: "Photo of candy grouped by colors."
+  caption: "Batching requests can reduce the number of client-server request/response cycles."
+  relative: false
+---
+
+With APISIX's request batching capability, you can group multiple API requests from the client into a single request. APISIX then turns this request into multiple requests to your upstream services, aggregates the response, and sends it back to the client as a single response.
+
+Batching can be useful in many practical scenarios where multiple requests are needed to fulfill a task. For example, the [Conference API](https://conferenceapi.azurewebsites.net/) exposes two different endpoints, `/speaker/{speakerId}/sessions` and `/speaker/{speakerId}/topics`, for listing the sessions and topics belonging to a speaker, respectively. You can make two API calls to show both in a front-end application.
+
+{{< figure src="/images/batching-requests/without-apisix.png#center" title="Making two requests" caption="Sometimes this might not be ideal." link="/images/batching-requests/without-apisix.png" target="_blank" class="align-center" >}}
+
+This might not always be ideal. Instead, you can use APISIX's request batching capability to group these requests into one, reducing the number of client-server calls.
+
+{{< figure src="/images/batching-requests/with-apisix.png#center" title="Batching requests" caption="Batching requests reduces the number of client-server calls." link="/images/batching-requests/with-apisix.png" target="_blank" class="align-center" >}}
```

Comment:
In the figures, I'd put the client on one side and APISIX and the upstreams close together on the other side to manifest the close network proximity.

You can also add a cloud between the client and APISIX to reinforce the network latency.